src/main/java/com/hartwig/platinum/kubernetes/KubernetesEngine.java

```diff
@@ -160,7 +150,7 @@ public KubernetesCluster findOrCreate(final String clusterName, final String run
             }
             return new KubernetesCluster(runName,
                     jobScheduler,
-                    new PipelineServiceAccountSecretVolume(jsonKey, kubernetesClientProxy, "service-account-key"),
+                    configuration.serviceAccount().get().kubernetesServiceAccount().get(),
```

Comment:
this looks problematic :) if these are really optionals you can use flatmaps/maps/orElse